File: www/src/lib/metastone/ActionType.js
/**
 * Created by chris on 1/17/17.
 */
import EnumExtended from '../EnumExtended';

export default class ActionType extends EnumExtended {
}

ActionType.initEnum([
    'SYSTEM',
    'END_TURN',
    'PHYSICAL_ATTACK',
    'SPELL',
    'SUMMON',
    'HERO_POWER',
    'BATTLECRY',
    'EQUIP_WEAPON',
    'DISCOVER',
]);

File: spellsource-server/src/main/java/com/hiddenswitch/framework/graphql/ActionType.java
package com.hiddenswitch.framework.graphql;

public enum ActionType {

    SYSTEM("SYSTEM"),
    END_TURN("END_TURN"),
    PHYSICAL_ATTACK("PHYSICAL_ATTACK"),
    SPELL("SPELL"),
    SUMMON("SUMMON"),
    HERO_POWER("HERO_POWER"),
    BATTLECRY("BATTLECRY"),
    EQUIP_WEAPON("EQUIP_WEAPON"),
    DISCOVER("DISCOVER"),
    HERO("HERO"),
    TAP("TAP");

    private final String graphqlName;

    private ActionType(String graphqlName) {
        this.graphqlName = graphqlName;
    }

    @Override
    public String toString() {
        return this.graphqlName;
    }

}
